docusaurus/docs/iOS/controllers/channels.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Channel Controllers
+---
+
+## ChannelListController
+
+`ChatChannelListController` allows you to observe a list of chat channels based on the provided query and to paginate channels.
+
+### ChannelList Delegate
+
+Classes that conform to the `ChatChannelListControllerDelegate` protocol will receive changes to the queried list of channels.
+
+```swift
+func controller(
+    _ controller: ChatChannelListController,
+    didChangeChannels changes: [ListChange<ChatChannel>]
+)
+```
+
+## ChannelController
+
+`ChatChannelController` allows you to observe and mutate data for one channel.
+
+### Channel Delegate
+
+Classes that conform to the `ChatChannelControllerDelegate` protocol will receive changes to channel data, members, messages and currently typing users.
+
+```swift
+func channelController(
+    _ channelController: ChatChannelController,
+    didUpdateChannel channel: EntityChange<ChatChannel>
+) {}
+
+func channelController(
+    _ channelController: ChatChannelController,
+    didUpdateMessages changes: [ListChange<ChatMessage>]
+) {}
+
+func channelController(
+    _ channelController: ChatChannelController,
+    didChangeTypingUsers typingUsers: Set<ChatUser>
+) {}
+
+func channelController(_ channelController: ChatChannelController, didReceiveMemberEvent: MemberEvent) {}
+```
+
+### Example: Typing Indicator
+
+Let's build a simple UIView that shows the current unread count for the current user.
+
+```swift
```

Comment:
This is a very large block of code. Consider breaking it into smaller blocks with prefacing content explaining the "Why" behind each of the steps or adding some comments above the two classes explaining what each does at a high level.